src/table.rs

```diff
@@ -153,7 +153,7 @@ impl<'db, 'txn, K: RedbKey + 'static, V: RedbValueMutInPlace + 'static> Table<'d
         if value_length as usize > MAX_VALUE_LENGTH {
             return Err(StorageError::ValueTooLarge(value_length as usize));
         }
-        let key_len = K::as_bytes(key.borrow()).as_ref().len();
+        let key_len = K::bytes_size(key.borrow());
```

Comment:
Instead of this new method, can you try moving these size checks to here: https://github.com/cberner/redb/blob/6e23b2fe86344b39e8502dc3a0c9bb3ac2803d50/src/tree_store/btree_base.rs#L666 ?
The data has already been serialized there, so it will avoid a second call into `RedbValue::as_bytes`

Reply:
Makes sense. Updated.

Reply:
I think you lost the change to insert_reserve() when you updated it

Reply:
:+1: